### PR TITLE
Add iterable to list messages in reverse order.

### PR DESCRIPTION
**Pull request by mihajenko**
Added a new API wrapper instance method for the List Messages command.

gitter.Gitter.listMessagesCursor: Makes a request for 100 (or less) message objects repeatedly, and yields each individual message object in reverse order (ie. newer message objects are first to be iterated over). Iterates until there is chat history to be consumed from the room.

### CODE REVIEW

**Comment by alexanderhiam** [comment]
Awesome! I left one comment - I've never actually used the github review system, so let me know if you can't find it....

**Comment by mihajenko** [comment]
Sorry, I can't find it. Just click on the + sign at the line you want and comment.

File: gitterpy/gitterpy.py [hunk 1]
```diff
@@ -84,6 +85,40 @@ def roomIdFromName(self, room_name):
...
+      sleep(sleep_time)
+
+      data = self._get(url_f.format(room_id, before_id))
+      if not data:
+        raise StopIteration()
+      before_id = data[0].get("id")
+      for message in reversed(data):
+        yield message
+
+    StopIteration()
```
Comment (alexanderhiam) [inline gitterpy/gitterpy.py]: Should this be `raise StopIteration()`?

**Comment by alexanderhiam** [comment]
Worth a try... comment added.

**Comment by mihajenko** [comment]
Great catch! I've committed the fix separately, but the newest commit tries to reuse a bit of code. Give this one a try.

**Comment by alexanderhiam** [comment]
Looks good to me. Thanks for the contribution!

**Pull request merged**